chalicelib/checks/audit_checks.py

```diff
@@ -378,6 +379,42 @@ def check_help_page_urls(connection, **kwargs):
     return check
 
 
+@check_function()
+def check_search_urls(connection, **kwargs):
+    '''Check the URLs in static sections that link to a search or browse page.
+    Give a warning if the number of results is 0.
+    '''
+    check = CheckResult(connection, 'check_search_urls')
+    search = ('search/?type=StaticSection' +
+              '&status%21=deleted&status%21=draft' +
+              '&field=body')
+    results = ff_utils.search_metadata(search, key=connection.ff_keys)
+    problematic_sections = {}
+    for result in results:
+        body = result.get('body', '')
+        # search links for search or browse pages, either explicit or relative
+        urls = re.findall(r'[\(|\[|=]["]*(?:[^\s\)\]]+(?:4dnucleome|elasticbeanstalk)[^\s\)\]]+|/)((?:browse|search)/\?[^\s\)\]]+)[\)|\]|"]', body)
+        if urls:
+            for url in urls:
+                url = re.sub(r'&limit=[^&]*|limit=[^&]*&?', '', url)  # remove limit if present
+                q_results = ff_utils.search_metadata(url + '&limit=1&field=@id', key=connection.ff_keys)
+                if len(q_results) == 0:
+                    problematic_sections[result['@id']] = problematic_sections.get(result['@id'], [])
```

Comment:
not something you need to change but just a future tip - you can also use `dict_name.setdefault(key_name, value)` if you need to check whether something is in the dictionary and create a new key if not.